xarray/core/computation.py

```diff
@@ -1721,3 +1727,59 @@ def _calc_idxminmax(
     res.attrs = indx.attrs
 
     return res
+
+
+def unify_chunks(*objects: T_DSorDA) -> Tuple[T_DSorDA, ...]:
+    """
+    Given any number of Dataset and/or DataArray objects, returns
+    new objects with unified chunk size along all chunked dimensions.
+
+    Returns
+    -------
+    unified (DataArray or Dataset) – Tuple of objects with the same type as
+    *objects with consistent chunk sizes for all dask-array variables
+
+    See Also
+    --------
+    dask.array.core.unify_chunks
+    """
+    from .dataarray import DataArray
+
+    # Convert all objects to datasets
+    datasets = [
+        obj._to_temp_dataset() if isinstance(obj, DataArray) else obj.copy()
+        for obj in objects
+    ]
+
+    # Get argumets to pass into dask.array.core.unify_chunks
+    unify_chunks_args = []
+    sizes: dict = {}
+    for ds in datasets:
+        for v in ds._variables.values():
+            if v.chunks is not None:
+                # Check sizes
+                sizes = {**ds.sizes, **sizes}
+                if not all(item in sizes.items() for item in ds.sizes.items()):
+                    raise ValueError(
+                        "Dimension lenghts are not consistent across chunked objects."
+                    )
```

Comment:
This feels unnecessarily hard to read. It's also unnecessarily O(n^2) when it could be O(n).
```suggestion
                for dim, size in ds.sizes.items():
                    prev_size = sizes.setdefault(dim, size)
                    if prev_size != size:
                        raise ValueError(f"Dimension {dim} size mismatch: {prev_size} != {size}")
```